.github/workflows/ci-unit-tests-report.yml

```diff
@@ -0,0 +1,41 @@
+name: Unit Test Results
+
+on:
+  workflow_run:
+    workflows: ["Unit Tests"]
+    types:
+      - completed
+
+jobs:
+  unit-test-results:
+    name: Unit Test Results
+    runs-on: ubuntu-latest
+
+    permissions:
+      checks: write
+
+      # Needed unless run with comment_mode: off.
+      pull-requests: write
+
+      # Required by download step to access artifacts API.
+      actions: read
+
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@eb238b55efaa70779f274895e782ed17c84f2895 # v2.6.1
+        with:
+          egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
+
+      - name: Download and Extract Artifacts
+        uses: dawidd6/action-download-artifact@v3
```

Comment:
do we know what's different about it?  It's preferable to use official actions like https://github.com/actions/download-artifact

Reply:
Not specifically, but I did spend a good number of hours trying out actions/download-artifact without any luck. Oddly, it wasn't able to discover the uploaded files, but there may have been a silly mistake on my part 🤷🏼 Happy to take suggestions!